common/geth/client.go

```diff
@@ -38,8 +38,8 @@ type EthClient struct {
 
 var _ common.EthClient = (*EthClient)(nil)
 
-func NewClient(config EthClientConfig, logger logging.Logger) (*EthClient, error) {
-	chainClient, err := ethclient.Dial(config.RPCURL)
+func NewClient(config EthClientConfig, rpcurl string, logger logging.Logger) (*EthClient, error) {
```

Comment:
It's confusing to take `EthClientConfig` which has a list of URLs and also to specify `rpcurl` in addition to that.
Since the only value we're using from `config` is `PrivateKeyString`, why don't we just take `privateKeyString` as input?
i.e. `NewClient(privateKeyString string, rpcurl string, logger logging.Logger)`

Reply:
the client also use NumConfirmations. To be future proof, pass in struct would be better. I will use an alterative to  to pass in an index.